src/client/opamCommands.ml

```diff
@@ -2406,65 +2407,67 @@ let switch cli =
       Arg.(some (list string)) None
   in
   let descr =
-    mk_opt ["description"] "STRING"
+    mk_opt ~cli cli_original ["description"] "STRING"
       "Attach the given description to a switch when creating it. Use the \
        $(i,set-description) subcommand to modify the description of an \
        existing switch."
       Arg.(some string) None
   in
   let full =
-    mk_flag ["full"]
+    mk_flag ~cli cli_original ["full"]
       "When exporting, include the metadata of all installed packages, \
        allowing to re-import even if they don't exist in the repositories (the \
        default is to include only the metadata of pinned packages)."
   in
   let freeze =
-    mk_flag ["freeze"]
+    mk_flag ~cli (cli_from cli2_1) ["freeze"]
       "When exporting, locks all VCS urls to their current commit, failing if \
        it can not be retrieved. This ensures that an import will restore the \
        exact state. Implies $(b,--full)."
   in
   let no_install =
-    mk_flag ["no-install"]
+    mk_flag ~cli cli_original ["no-install"]
       "When creating a local switch, don't look for any local package \
        definitions to install."
   in
   let deps_only =
-    mk_flag ["deps-only"]
+    mk_flag ~cli cli_original ["deps-only"]
       "When creating a local switch in a project directory (i.e. a directory \
        containing opam package definitions), install the dependencies of the \
        project but not the project itself."
   in
   let force =
-    mk_flag ["force"]
+    mk_flag ~cli (cli_from cli2_1) ["force"]
       "Only for $(i,set-invariant): force setting the invariant, bypassing \
        consistency checks."
   in
   let no_action =
-    mk_flag ["n"; "no-action"]
+    mk_flag ~cli (cli_from cli2_1) ["n"; "no-action"]
       "Only for $(i,set-invariant): set the invariant, but don't enforce it \
        right away: wait for the next $(i,install), $(i,upgrade) or similar \
        command."
   in
   (* Deprecated options *)
   let d_alias_of =
-    mk_opt ["A";"alias-of"]
-      "COMP"
-      "This option is deprecated."
-      Arg.(some string) None
+    mk_opt ~cli (cli_between cli2_1 cli2_2)
+      ["A";"alias-of"] "COMP" "Deprecated" Arg.(some string) None
```

Comment:
```suggestion
    mk_opt ~cli (cli_between cli2_0 cli2_1)
      ["A";"alias-of"] "COMP" "Deprecated." Arg.(some string) None
```

Reply:
Those option were removed in opam 2.0 and reintroduced with a deprecation notice in 2.1. (cf #3685). We could set theme as available in cli2.0 and removed in 2.1, from the cli point of view, but it is not correct according the real code, where the option results on an error with opam 2.0.x series, and a deprecation message since the 2.1.0~alpha.

Reply:
Those three options were reintroduced in 2.0.2 (see https://github.com/ocaml/opam/pull/3669). So while 2.0.0 and 2.0.1 don't have them, the latest service release of 2.0 does. I think it's OK to change that behaviour to the better "gone in 2.1" (it's a beta rather than an RC)

Reply:
ok, lgtm